src/routes/process-user-vocabulary/components/AudioRecordOverlay.tsx

```diff
@@ -0,0 +1,153 @@
+  border-radius: 100px;
+  background-color: ${props =>
+    props.isPressed ? props.theme.colors.audioRecordingActive : props.theme.colors.disabled};
+`
+
+const Content = styled.View`
+  margin-top: ${hp('12%')}px;
+  margin-bottom: ${hp('6%')}px;
+  flex: 1;
+  justify-content: space-between;
+`
+
+const Heading = styled(HeadingText)`
+  text-align: center;
+  font-size: ${props => props.theme.spacings.lg};
+`
+
+const HeadingContainer = styled.View`
+  min-height: ${hp('12%')}px;
+`
+
+const RecordingInfo = styled.Text`
+  font-size: ${props => props.theme.spacings.lg};
+  text-align: center;
+`
+
+const MeteringInfo = styled.View`
+  flex-direction: row;
+  align-self: center;
+  min-height: ${hp('12%')}px;
+`
+
+const MeteringBar = styled.View<{ height: number }>`
+  min-height: ${props => props.theme.spacings.xxs}
+  height: ${props => `${props.height * 2}`}px;
+  width: ${props => props.theme.spacings.xxs}
+  background-color: ${props => props.theme.colors.audioRecordingActive};
+  border-radius-: 50px;
+  align-self: center;
+  margin: 0 1px;
+`
+
+const InfoContainer = styled.View`
+  justify-content: flex-end;
+`
+
+interface AudioRecordOverlayProps {
+  onClose: () => void
+  onStartRecording: () => Promise<void>
+  onStopRecording: () => Promise<void>
+  recordingTime: string
+  meteringResults: number[]
+}
+
+// Zero alignment of values with the minimum metering value
+const cleanUpMeteringResults = (meteringResults: number[]): number[] => {
+  const filteredResults = meteringResults.filter(el => el !== Math.min(...meteringResults))
+  return filteredResults.map(el => el + Math.abs(Math.min(...filteredResults)))
+}
+
+const AudioRecordOverlay = ({
+  onClose,
+  onStartRecording,
+  onStopRecording,
+  recordingTime,
+  meteringResults,
+}: AudioRecordOverlayProps): ReactElement => {
+  const [isPressed, setIsPressed] = useState<boolean>(false)
+  const permissionGranted = useGrantPermissions(
+    Platform.OS === 'android'
+      ? [
+          PERMISSIONS.ANDROID.RECORD_AUDIO,
+          PERMISSIONS.ANDROID.READ_EXTERNAL_STORAGE,
+          PERMISSIONS.ANDROID.WRITE_EXTERNAL_STORAGE,
+        ]
+      : PERMISSIONS.IOS.MICROPHONE
+  )
+  const { hold, talk } = getLabels().general.audio
+  const theme = useTheme()
+
+  const cleanedMetering = useMemo(() => cleanUpMeteringResults(meteringResults), [meteringResults])
+
+  return (
+    <RNModal visible transparent animationType='fade' onRequestClose={() => onClose()}>
+      <Container>
+        <Icon onPress={() => onClose()}>
+          <CloseIcon width={theme.spacingsPlain.lg} height={theme.spacingsPlain.lg} />
+        </Icon>
+        <Content>
+          <HeadingContainer>
+            <Heading>{isPressed ? talk : hold}</Heading>
+          </HeadingContainer>
+          <InfoContainer>
+            <MeteringInfo>
+              {cleanedMetering.map((element, index) => (
+                // eslint-disable-next-line react/no-array-index-key -- no better key available
```

Comment:
I am not really happy with this, as I think linting rules should not be disabled and this rule does really prevent bug, but in this case there should not occure a bug, but i think this can be improved.
Please add e.g. a timestamp or id for each bar.

Reply:
unfortunately timestamp also generate key duplicate so i decided to use uuid generator